### PR TITLE
[FEATURE] Refactor issues templates and add a new one

### PR DESCRIPTION
**Pull request by kilian-goetz**
## 1) Description

`ISSUE_TEMPLATE.md` and `FEATURE_REQUEST_TEMPLATE.md` were changed in other format `yml` with more feature (explain in next part).
A new template for refactor was added `REFACTOR_TEMPLATE.yml`.

## 2) Technical choice

I changed format of issues templates from `.md` to `.yml`. This format includes more relevant feature, in fact we can enforce to fill parts or checkboxes. If you want further information, you can refer to [GitHub documentation](https://docs.github.com/fr/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms).

## 3) Checks

- [x] My code follows the contributing guidelines
- [x] I have read the code of conduct
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/ISSUE_TEMPLATE.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      required: true
+  - type: textarea
+    attributes:
+      label: Steps to Reproduce
+      description: Please provide detailed steps for reproducing the issue.
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Expected Behavior
```
Comment (galyfray) [inline .github/ISSUE_TEMPLATE/ISSUE_TEMPLATE.yml:19]: There are two expected behavior. One should be enough

File: .github/ISSUE_TEMPLATE/REFACTOR_TEMPLATE.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: 🏭 Refactor request
+description: Refactor an existing feature
+title: "[Bug]: "
```
Comment (galyfray) [inline .github/ISSUE_TEMPLATE/REFACTOR_TEMPLATE.yml]:
```suggestion
title: "[REFACTOR]: "
```

File: .github/ISSUE_TEMPLATE/ISSUE_TEMPLATE.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+name: 🐛 Bug Report
+description: Report an unexpexted behaviour
+title: "[Bug]: "
```
Comment (galyfray) [inline .github/ISSUE_TEMPLATE/ISSUE_TEMPLATE.yml]:
```suggestion
title: "[BUG]: "
```

File: .github/ISSUE_TEMPLATE/REFACTOR_TEMPLATE.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+labels:
+  - Refactor
```
Comment (galyfray) [inline .github/ISSUE_TEMPLATE/REFACTOR_TEMPLATE.yml]:
```suggestion
  - refactor
```

**Review by galyfray — CHANGES_REQUESTED** [review]
I think too many fields are required in the bug report, I think that some of them might be optional.

File: .github/ISSUE_TEMPLATE/ISSUE_TEMPLATE.yml (hunk 1 above)
Comment (kilian-goetz) [reply .github/ISSUE_TEMPLATE/ISSUE_TEMPLATE.yml:19]: Oops, it's a mistake.

**Comment by kilian-goetz** [comment]
> I think too many fields are required in the bug report, I think that some of them might be optional.

Can you tell me which field you want mandatory and not ?

Other comments were resolved.

**Review by galyfray — APPROVED** [review]

**Pull request merged**